### DISCUSSION TITLE
自定义快捷键无法覆盖原来的快捷键

### DISCUSSION THREAD

**Discussion by 834116404** [sublimetext-io/docs.sublimetext.io#81] [Q&A]
![image](https://github.com/sublimetext-io/docs.sublimetext.io/assets/36099756/d98d0ef4-c540-4e8f-a704-11488d1d9b95)

配置了 alt +d  然后无法覆盖默认的 { "keys": ["ctrl+shift+k"], "command": "run_macro_file", "args": {"file": "res://Packages/Default/Delete Line.sublime-macro"} }, 这个配置..

**Comment by jfcherng** [comment]
ST never deletes/overwrites a keybinding. You can image that it just puts your custom keybindings before defaults. So `ctrl+shift+k` still runs the same command.

Yet I don't know what's your expectation honestly.

**Reply by 834116404** [reply]
我期望是 我想 alt +d 配置为删除当前行,  但是配置完了发现不管用

**Comment by 834116404** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        
            
        
    
    我现在想配置 alt + d 删除整行, 但是发现下面的配置无效, 请问是什么原因?[		{ "keys": ["alt+d","alt+k","ctrl+k"], "command": "run_macro_file", "args": {"file": "res://Packages/Default/Delete Line.sublime-macro"} }]
    
      
    
        
                    
            
                
                    
                            
                                
                            
                            
                                张俊杰
                            
                    
                        
                            
                                    
                                    ***@***.***
                                    
                                    
                                    
                            
                        
                
            
        
        
    
 
    

  
    ---- 回复的原邮件 ----
  
  
    
      
        
         发件人 
        
        
            Jack ***@***.***>
            
        
      
      
        
         发送日期 
        
        
        2023年09月26日 17:38
        
      
      
        
         收件人 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         抄送人 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         主题 
        
        
              Re: [sublimetext-io/docs.sublimetext.io] 自定义快捷键无法覆盖原来的快捷键 (Issue #80)
        
      
    
  
  
ST never deletes/overwrites a keybinding. You can image it just put your custom keybinding before
defaults.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by jfcherng** [comment]
Get super confused.

You say you want `alt+d` while you actually use `"alt+d","alt+k","ctrl+k"`.

**Reply by 834116404** [reply]
我是想 alt +d 去配置 删除当前行, 结果发现不管用, 我又尝试着配置 "alt+d","alt+k","ctrl+k" 为删除当前行, 发现也不管用..

**Comment by jfcherng** [comment]
> 我是想 alt +d 去配置 删除当前行, 结果发现不管用, 我又尝试着配置 "alt+d","alt+k","ctrl+k" 为删除当前行, 发现也不管用..

Your config works fine for me so you would have to debug this from the very beginning.

1. Does ST think you press `alt+d`?

   Execute `sublime.log_input(True)` in ST console (``ctrl+` ``) and then press `alt+d`. What's the console  output?

2. Have the wanted command been fired?

   Execute `sublime.log_commands(True)` in ST console (``ctrl+` ``) and then try to trigger your keybinding. What's the console output?

**Reply by 834116404** [reply]
谢谢,弄好了, 我直接将原来的所有的快捷键配置都复制过来 , 然后将 alt +d 改成别的, 然后 再给 delete line 改成 alt + d   ,发现好使了.
